src/Database/LSMTree/Internal/MergingRun.hs

```diff
@@ -789,27 +819,24 @@ remainingMergeDebt (DeRef mr) = do
 -- supplied credits. See 'supplyCreditsAbsolute' to set the supplied credits
 -- to an absolute value.
 --
--- The result is:
---
---  1. The (absolute value of the) supplied credits beforehand.
---  2. The (absolute value of the) supplied credits afterwards.
---  3. The number of credits left over. This will be non-zero if the credits
---     supplied would take the total supplied credits over the total merge debt.
+-- The result is the number of credits left over. This will be non-zero if the
+-- credits supplied would take the total supplied credits over the total merge
+-- debt.
 --
 supplyCreditsRelative ::
      forall t m h. (MonadSTM m, MonadST m, MonadMVar m, MonadMask m)
   => Ref (MergingRun t m h)
   -> CreditThreshold
   -> MergeCredits
-  -> m (MergeCredits, MergeCredits, MergeCredits)
-       -- ^ (suppliedCredits, suppliedCredits', leftoverCredits)
-supplyCreditsRelative mr th c = do
-    r@(_suppliedCredits, suppliedCredits', leftoverCredits)
-      <- supplyCredits mr th (SupplyMergeCredits SupplyRelative c)
-
-    assert (suppliedCredits' == mergeDebtAsCredits (totalMergeDebt mr)
-            || leftoverCredits == 0) $
-      pure r
+  -> m MergeCredits
```

Comment:
The reason we were returning all three `MergeCredits` values was so that in the `MergeSchedule` we would have the opportunity to trace the before/after credits. I still think that'd be useful, though we've not done it yet.

Reply:
Oh I see, right. I'd forgotten about the absolute/relative and which ones were used where. Yes that's fine.